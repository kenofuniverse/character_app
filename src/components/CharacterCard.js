import React from 'react'
import PropTypes from 'prop-types'
import {
  Card
} from 'react-bootstrap'
import { Link } from 'react-router-dom'
import * as moment from 'moment'

const fromNow = (date) => moment(date).fromNow()

const CharacterCard = ({item}) => (
  <Card>
    <Card.Img variant="top" src={item.image} />
    <Card.Body>
      <Card.Title>
        <Link to={`/character/${item.id}`}>{item.name}</Link>
      </Card.Title>
      <Card.Text>
        {item.type}
      </Card.Text>
    </Card.Body>
    <Card.Footer>
      <small className="text-muted">{fromNow(item.created)}</small>
    </Card.Footer>
  </Card>
)

CharacterCard.propTypes = {
  item: PropTypes.object.isRequired,
}

export default CharacterCard
