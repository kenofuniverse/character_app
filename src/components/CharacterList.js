import React from 'react'
import PropTypes from 'prop-types'
import { CardColumns } from 'react-bootstrap'
import CharacterCard from './CharacterCard'

const CharacterList = ({ items }) => {
  return (
    <CardColumns>
      {items.map(item => (
        <CharacterCard key={item.id} item={item} />
      ))}
    </CardColumns>
  )
}

CharacterList.defaultProps = {
  items: []
}

CharacterList.propTypes = {
  items: PropTypes.array
}

export default CharacterList
