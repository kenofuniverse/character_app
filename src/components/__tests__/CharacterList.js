import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router';
import CharacterCard from '../CharacterCard'
import CharacterList from '../CharacterList'

describe('CharacterDetail component', () => {
  const items = [
    {
      "id": 1,
      "name": "Rick Sanchez",
      "status": "Alive",
      "species": "Human",
      "type": "",
      "gender": "Male",
      "origin": {
        "name": "Earth (C-137)",
        "url": "https://rickandmortyapi.com/api/location/1"
      },
      "location": {
        "name": "Earth (Replacement Dimension)",
        "url": "https://rickandmortyapi.com/api/location/20"
      },
      "image": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
      "episode": [
        "https://rickandmortyapi.com/api/episode/1",
        "https://rickandmortyapi.com/api/episode/2",
        // ...
      ],
      "url": "https://rickandmortyapi.com/api/character/1",
      "created": "2017-11-04T18:48:46.250Z"
    },
    {
      "id": 183,
      "name": "Johnny Depp",
      "status": "Alive",
      "species": "Human",
      "type": "",
      "gender": "Male",
      "origin": {
        "name": "Earth (C-500A)",
        "url": "https://rickandmortyapi.com/api/location/23"
      },
      "location": {
        "name": "Earth (C-500A)",
        "url": "https://rickandmortyapi.com/api/location/23"
      },
      "image": "https://rickandmortyapi.com/api/character/avatar/183.jpeg",
      "episode": [
        "https://rickandmortyapi.com/api/episode/8"
      ],
      "url": "https://rickandmortyapi.com/api/character/183",
      "created": "2017-12-29T18:51:29.693Z"
    }
  ]

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <CharacterList items={items} />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders contents correctly', () => {
    const wrapper = shallow(<CharacterList items={items} />)
    expect(wrapper.find(CharacterCard)).toHaveLength(items.length)
  })
})
