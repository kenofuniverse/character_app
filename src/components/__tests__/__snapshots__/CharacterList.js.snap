// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetail component renders correctly 1`] = `
<div
  className="card-columns"
>
  <div
    className="card card"
  >
    <img
      className="card-img-top card__image"
      src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
    />
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      >
        <a
          href="/character/1"
          onClick={[Function]}
        >
          Rick Sanchez
        </a>
      </div>
      <p
        className="card-text"
      >
        
      </p>
    </div>
    <div
      className="card-footer"
    >
      <div
        className="row"
      >
        <div
          className="col-8"
        >
          <small
            className="text-muted"
          >
            Created 
            a year ago
          </small>
        </div>
        <div
          className="text-right col-4"
        >
          <div
            className="form-check"
          >
            <input
              checked={false}
              className="form-check-input position-static"
              disabled={false}
              onChange={[Function]}
              type="checkbox"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card card"
  >
    <img
      className="card-img-top card__image"
      src="https://rickandmortyapi.com/api/character/avatar/183.jpeg"
    />
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      >
        <a
          href="/character/183"
          onClick={[Function]}
        >
          Johnny Depp
        </a>
      </div>
      <p
        className="card-text"
      >
        
      </p>
    </div>
    <div
      className="card-footer"
    >
      <div
        className="row"
      >
        <div
          className="col-8"
        >
          <small
            className="text-muted"
          >
            Created 
            a year ago
          </small>
        </div>
        <div
          className="text-right col-4"
        >
          <div
            className="form-check"
          >
            <input
              checked={false}
              className="form-check-input position-static"
              disabled={false}
              onChange={[Function]}
              type="checkbox"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
